File: java/base/src/main/vsdk/toolkit/common/symbolicAlgebra/_AlgebraicExpressionConstantNode.java
package vsdk.toolkit.common.symbolicAlgebra;
import java.io.Serial;
import vsdk.toolkit.common.VSDK;

public class _AlgebraicExpressionConstantNode extends _AlgebraicExpressionNode
{
    @Serial private static final long serialVersionUID = 20071014L;

    private double val;
    public _AlgebraicExpressionConstantNode(double val)
    {
        this.val = val;
    }

    @Override
    public double eval() throws AlgebraicExpressionException
    {
        return val;
    }

    @Override
    public String toString()
    {
        String msg;

        msg = "" + VSDK.formatDouble(val);
        return msg;
    }
}

File: typescript/base/src/main/vsdk/toolkit/common/symbolicAlgebra/_AlgebraicExpressionConstantNode.ts
import { VSDK } from "../VSDK.js";
import { _AlgebraicExpressionNode } from "./_AlgebraicExpressionNode.js";
export class _AlgebraicExpressionConstantNode extends _AlgebraicExpressionNode {
    public constructor(private readonly value: number) {
        super();
    }
    public eval(): number {
        return this.value;
    }
    public override toString(): string {
        return VSDK.formatDouble(this.value);
    }
}
